### PR TITLE
Adding x86 CI build and test run.

### LINKED ISSUE #13448: Add Windows 32 bit to test build
We are currently not testing 32 bit. (Which we support on Windows)

We need to enable this for the test build. Separately we should enable it for CI -- @mmitche what is involved to do that, once the test build is going?

/cc @karelz @weshaggard

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @danmosemsft @mmitche 

Changing WIndows Debug build to run x86 build and test instead.

Fixes #13448

### CODE REVIEW

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by weshaggard** [comment]
@joperezr have we fulled out any issues with running our tests for x86? Have we created an official test run for it yet?

**Comment by joperezr** [comment]
No official test run for it yet, that is my next task. And at least locally I have the x86 test running fine, I'm going to queue a CI job with the produced x86 definition and make sure it's green before I merge this

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
Here are the builds that I triggered with the new definitions:
[Windows Debug x86](https://ci.dot.net/job/dotnet_corefx/job/master/job/GenPRTest/job/windows_nt_debug_prtest/1/)
[Windows Release x64](https://ci.dot.net/job/dotnet_corefx/job/master/job/GenPRTest/job/windows_nt_release_prtest/1/)

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test please (Jenkins never started the leg)

**Comment by weshaggard** [comment]
The x86 run results have been cleaned up so I cannot look at them right now but when I quickly looked at them yesterday there appeared to be failures. Please make sure they are clean before merging this.

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
> Please make sure they are clean before merging this.

I have re generated and requeue builds for those two definitions to analyze failures. links above have been updated and should take you know to the corresponding builds

**Comment by joperezr** [comment]
@weshaggard both of my generated runs passed after removing the layout project, is this good to merge?

**Comment by weshaggard** [comment]
Yes I'm good with it. Thanks.

File: netci.groovy [hunk 1]
```diff
@@ -359,7 +364,7 @@ def osShortName = ['Windows 10': 'win10',
                 // On Windows we use the packer to put together everything. On *nix we use tar
                 steps {
                     if (osName == 'Windows 10' || osName == 'Windows 7' || osName == 'Windows_NT') {
-                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd -${configurationGroup} -os=${osGroup} -- /p:WithoutCategories=IgnoreForCI")
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd -${configurationGroup} -os:${osGroup} -buildArch:${buildArchConfiguration[configurationGroup]} -TestNugetRuntimeId:${testNugetRuntimeIdConfiguration[configurationGroup]} -- /p:WithoutCategories=IgnoreForCI")
```
Comment (joperezr) [inline netci.groovy]: I actually thought of that but didn't know where is that configured, I'll change that as well.

**Pull request merged**